### PR TITLE
feat(core) add testing and refactor

### PR DESCRIPTION
**Pull request by tensor-programming**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [ ] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

More refactoring and test coverage.  Refactoring in service of making testing easier and making rust more idiomatic.

### CODE REVIEW

**Comment by tensor-programming** [comment]
Not finished but open for review.

File: tauri/src/app/runner.rs [hunk 1]
```diff
@@ -200,3 +200,80 @@ fn build_webview(
...
+  fn init_config() -> crate::config::Config {
+    crate::config::get().expect("unable to setup default config")
+  }
+
+  #[test]
+  fn check_setup_content() {
+    let config = init_config();
+    let c = config.clone();
+
+    if cfg!(not(any(feature = "embedded-server", feature = "no-server"))) {
```
Comment (lucasfernog) [inline tauri/src/app/runner.rs]: this block could be the final "else"

Comment (tensor-programming) [reply tauri/src/app/runner.rs]: I may just remove the entire cfg! logic.  There is a way to shrink it considering the main difference is the check afterwards.

**Comment by tensor-programming** [comment]
Going to merge and then Ill do another pass to build out more tests.

**Pull request merged**